lib/db/schema/patch-022-023.sql

```diff
@@ -0,0 +1,142 @@
+-- This migration:
+--
+-- * Changes the type of the devices.callbackPublicKey column
+-- * Add a new devices.callbackAuthKey column
+-- * Updates the accountDevices stored procedure with the new column
+-- * Alters three stored procedures, createDevice, updateDevice and
+--   sessionWithDevice to work with the new columns
+
+-- 2-steps callbackPublicKey change: we avoid junk values, and we don't block
+-- concurent queries.
+
+ALTER TABLE devices
+CHANGE callbackPublicKey oldCallbackPublicKey BINARY(32),
+ADD COLUMN callbackPublicKey CHAR(88),
+ADD COLUMN callbackAuthKey CHAR(24),
+ALGORITHM = INPLACE, LOCK = NONE;
+
+ALTER TABLE devices
+DROP COLUMN oldCallbackPublicKey,
+ALGORITHM = INPLACE, LOCK = NONE;
+
```

Comment:
oh, sorry, I thought the `ALGORITHM = INPLACE, LOCK = NONE;` stuff would help ensure this went smoothly - can you say more about how it fails?